gluten-substrait/src/main/scala/org/apache/gluten/config/GlutenConfig.scala

```diff
@@ -963,7 +963,7 @@ object GlutenConfig {
       .createWithDefault(100000)
 
   val COLUMNAR_TABLE_CACHE_ENABLED =
-    buildConf("spark.gluten.sql.columnar.tableCache")
+    buildStaticConf("spark.gluten.sql.columnar.tableCache")
```

Comment:
I suspect this option is somehow leaked among different test suites so caused some strange errors. Let's correct it as static to align with its life scope and doing this will also help diagnosing the issue.